CRM/Contribute/Form/ContributionBase.php

```diff
@@ -353,7 +353,7 @@ public function preProcess() {
 
       $this->_paymentProcessorIDs = array_filter(explode(
         CRM_Core_DAO::VALUE_SEPARATOR,
-        CRM_Utils_Array::value('payment_processor', $this->_values)
+        ($this->_values['payment_processor'] ?? 0)
```

Comment:
This feels like it might be a material change - I'm nos sure - just one that seems like it might need more than a quick eyeball

Reply:
@eileenmcnaughton I'm not sure 

```
seamus@civicrmdevelopment:~/buildkit/build/47-test/web/sites/all/modules/civicrm$ cv ev "var_dump(array_filter(explode(',', NULL)));"
array(0) {
}

seamus@civicrmdevelopment:~/buildkit/build/47-test/web/sites/all/modules/civicrm$ cv ev "var_dump(array_filter(explode(',', 0)));"
array(0) {
}

seamus@civicrmdevelopment:~/buildkit/build/47-test/web/sites/all/modules/civicrm$
```

@demeritcowboy do you have an opinion on this?

Reply:
Your comparison makes sense, just that I expect at some point php to start fully enforcing data types, so calling explode on int 0 seems off - maybe should be `''`?

Reply:
ok updated to that @demeritcowboy and confirmed it produces same output

```
cv ev "var_dump(array_filter(explode(',', '')));"
array(0) {
}
```